### PR TITLE
Challenge2 | sqs-processor

### PR DESCRIPTION
**Pull request by CristinaHA93**
There are tests for TO DO

### CODE REVIEW

**Review by tudorcodrea — APPROVED** [review]

File: letter-processor/src/main/java/com/christmas/letter/processor/mapper/LetterMapper.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.christmas.letter.processor.mapper;
+
+import com.christmas.letter.processor.dto.LetterDto;
+import com.christmas.letter.processor.entities.Letter;
+import org.mapstruct.Mapper;
+
+@Mapper
```
Comment (teodora-bobirneci) [inline letter-processor/src/main/java/com/christmas/letter/processor/mapper/LetterMapper.java]: If you use the annotation like this @Mapper(componentModel = "spring"), you don't need the MapperConfiguration class anymore.

File: letter-processor/src/test/java/com/christmas/letter/processor/service/LetterServiceProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  @Mock
+  private LetterMapper letterMapper;
+
+  @InjectMocks
+  private LetterServiceProcessor letterService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
```
Comment (teodora-bobirneci) [inline letter-processor/src/test/java/com/christmas/letter/processor/service/LetterServiceProcessorTest.java]: You are already using @ExtendWith(MockitoExtension.class), you shouldn't need this openMocks call here.

File: letter-processor/src/test/java/com/christmas/letter/processor/service/LetterServiceProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    LetterDto result = letterService.getLetterByEmail(email);
+
+    // Assert
+    assertNotNull(result);
+    verify(letterRepository, times(1)).findById(email);
+
+  }
+
+
+    // TO DO error tests and getAllLetter tests
```
Comment (teodora-bobirneci) [inline letter-processor/src/test/java/com/christmas/letter/processor/service/LetterServiceProcessorTest.java]: Agreed :-) also you need integration tests for both methods, negative scenarios, pagination etc.

File: letter-processor/src/main/java/com/christmas/letter/processor/service/LetterServiceProcessor.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import com.christmas.letter.processor.repository.LetterRepository;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class LetterServiceProcessor {
```
Comment (teodora-bobirneci) [inline letter-processor/src/main/java/com/christmas/letter/processor/service/LetterServiceProcessor.java]: This class' name should be LetterProcessorService - the convention is to put the "Service" part at the end.

File: letter-processor/src/main/java/com/christmas/letter/processor/controller/LetterProcessorController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+public class LetterProcessorController {
+
+  @NonNull
+  private final LetterServiceProcessor service;
+
+  @GetMapping("/letters/{email}")
+  public ResponseEntity<LetterDto> getLetterByEmail(@PathVariable String email) {
+
+    LetterDto letter = service.getLetterByEmail(email);
+    if (letter == null) {
```
Comment (teodora-bobirneci) [inline letter-processor/src/main/java/com/christmas/letter/processor/controller/LetterProcessorController.java]: Instead of doing this check here, you could do the verification in the service layer, and throw NotFoundException if the letter doesn't exist. Then you can define a global exception handler where you handle the NotFoundException (AOP).

File: letter-processor/src/main/java/com/christmas/letter/processor/mapper/LetterMapper.java (hunk 1 above)
Comment (CristinaHA93) [reply letter-processor/src/main/java/com/christmas/letter/processor/mapper/LetterMapper.java]: used componentModel = "spring" and remove MapperConfiguration

File: letter-processor/src/test/java/com/christmas/letter/processor/service/LetterServiceProcessorTest.java (hunk 3 above)
Comment (CristinaHA93) [reply letter-processor/src/test/java/com/christmas/letter/processor/service/LetterServiceProcessorTest.java]: I will continue with tests

File: letter-processor/src/main/java/com/christmas/letter/processor/service/LetterServiceProcessor.java (hunk 4 above)
Comment (CristinaHA93) [reply letter-processor/src/main/java/com/christmas/letter/processor/service/LetterServiceProcessor.java]: the class was renamed in LetterProcessorService

File: letter-processor/src/main/java/com/christmas/letter/processor/controller/LetterProcessorController.java (hunk 5 above)
Comment (CristinaHA93) [reply letter-processor/src/main/java/com/christmas/letter/processor/controller/LetterProcessorController.java]: I removed this check because in service I checked that NotFoundException
